pkg/controller/install/install_controller.go

```diff
@@ -161,3 +162,36 @@ func autoInstall(c client.Client, ns string) (err error) {
 	}
 	return err
 }
+
+func matchesClusterRole(roles []string, cr string) bool {
+	for _, i := range roles {
+		if cr == i {
+			return true
+		}
+	}
+	return false
+}
+
+func addSCCforSpecialClusterRoles(u *unstructured.Unstructured) *unstructured.Unstructured {
+
+	// these do need some openshift specific SCC
+	clusterRoles := []string{
+		"eventing-broker-filter",
+		"knative-eventing-controller",
+		"in-memory-channel-controller",
+		"in-memory-channel-dispatcher",
+	}
+
```

Comment:
Given that Go provides no convenient `find` function and your roles are hard-coded anyway, I might move `matchesClusterRole` here where `clusterRoles` is in scope, e.g.
```suggestion
matchesClusterRole := func(cr string) bool {
	for _, i := range clusterRoles {
		if cr == i {
			return true
		}
	}
	return false
}
```